src/passes/referencedLibraries.ts

```diff
@@ -0,0 +1,34 @@
+import assert = require('assert');
+import { ContractDefinition, ContractKind, FunctionCall, MemberAccess } from 'solc-typed-ast';
+import { AST } from '../ast/ast';
+import { ASTMapper } from '../ast/mapper';
+
+export class ReferencedLibraries extends ASTMapper {
+  visitFunctionCall(node: FunctionCall, ast: AST): void {
+    if (node.vExpression instanceof MemberAccess) {
+      const contractDef = new Map();
+
+      node.vExpression.context?.map.forEach((astNode, key) => {
+        if (astNode instanceof ContractDefinition && astNode.kind === ContractKind.Library) {
+          contractDef.set(key, astNode);
+        }
+      });
+
+      const func_id = node.vExpression.referencedDeclaration;
+
+      contractDef.forEach((astNode, key) => {
```

Comment:
This is clever, and as far as I can tell a completely valid way of doing this, but a comment block explaining how the function works would aid maintainability I think